src/test/WormholeJoin.t.sol

```diff
@@ -0,0 +1,569 @@
+        balanceOf[src] = sub(balanceOf[src], wad);
+        balanceOf[dst] = add(balanceOf[dst], wad);
+        return true;
+    }
+    function mint(address usr, uint256 wad) external  {
+        balanceOf[usr] = add(balanceOf[usr], wad);
+    }
+    function burn(address usr, uint256 wad) external {
+        require(balanceOf[usr] >= wad, "Dai/insufficient-balance");
+        if (usr != msg.sender && allowance[usr][msg.sender] != type(uint256).max) {
+            require(allowance[usr][msg.sender] >= wad, "Dai/insufficient-allowance");
+            allowance[usr][msg.sender] = sub(allowance[usr][msg.sender], wad);
+        }
+        balanceOf[usr] = sub(balanceOf[usr], wad);
+    }
+    function approve(address usr, uint256 wad) external returns (bool) {
+        allowance[msg.sender][usr] = wad;
+        return true;
+    }
+}
+
+contract DaiJoinMock {
+    VatMock public vat;
+    DaiMock public dai;
+
+    constructor(address vat_, address dai_) {
+        vat = VatMock(vat_);
+        dai = DaiMock(dai_);
+    }
+    uint256 internal constant RAY = 10 ** 27;
+    function mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x);
+    }
+    function join(address usr, uint256 wad) external {
+        vat.move(address(this), usr, mul(RAY, wad));
+        dai.burn(msg.sender, wad);
+    }
+    function exit(address usr, uint256 wad) external {
+        vat.move(msg.sender, address(this), mul(RAY, wad));
+        dai.mint(usr, wad);
+    }
+}
+
+contract WormholeJoinTest is DSTest {
+
+    Hevm internal hevm = Hevm(HEVM_ADDRESS);
+    WormholeJoin internal join;
+    VatMock internal vat;
+    DaiMock internal dai;
+    DaiJoinMock internal daiJoin;
+    address internal vow = address(111);
+
+    uint256 internal constant RAD = 10**45;
+
+    function setUp() public {
+        vat = new VatMock();
+        dai = new DaiMock();
+        daiJoin = new DaiJoinMock(address(vat), address(dai));
+        join = new WormholeJoin(address(vat), address(daiJoin), "L2DAI", "ethereum");
+        join.file("line", "l2network", 1_000_000 ether);
+        join.file("vow", vow);
+        join.file("fees", "l2network", address(new WormholeConstantFee(0)));
+        vat.hope(address(daiJoin));
+    }
+
+    function _ink() internal view returns (uint256 ink_) {
+        (ink_,) = vat.urns(join.ilk(), address(join));
+    }
+
+    function _art() internal view returns (uint256 art_) {
+        (, art_) = vat.urns(join.ilk(), address(join));
+    }
+
+    function _blessed(WormholeGUID memory guid) internal view returns (bool blessed_) {
+        (blessed_, ) = join.wormholes(getGUIDHash(guid));
+    }
+
+    function _pending(WormholeGUID memory guid) internal view returns (uint248 pending_) {
+        (, pending_) = join.wormholes(getGUIDHash(guid));
+    }
+
+    function _tryRely(address usr) internal returns (bool ok) {
+        (ok,) = address(join).call(abi.encodeWithSignature("rely(address)", usr));
```

Comment:
I think using `encodeWithSelector(...)` is a little more robust against breaking changes. Same with the trys below.

Reply:
Hmm I feel that it is clearer for the tests use this one and not following which type of braking change could happen here.

Reply:
Ok, I don't feel strongly about this. So good to merge I think.